### PR TITLE
 Use the new child specifications in the Supervisor module

### PR DESCRIPTION
**Pull request by hseinmoussa**

### CODE REVIEW

**Comment by danhawkins** [comment]
@hseinmoussa can you bump the version also, so we can publish to hex

File: lib/ex_schedule.ex [hunk 1]
```diff
@@ -100,7 +100,7 @@ defmodule ExSchedule do
 
       @doc false
       def init(opts) do
-        schedules() |> Enum.map(&child_spec/1) |> supervise(strategy: :one_for_one)
+        schedules() |> start() |> Supervisor.init(strategy: :rest_for_one)
```
Comment (zorbash) [inline lib/ex_schedule.ex:103]: [question] Why the change to `rest_for_one`? 
With `rest_for_one`:

> if a child process terminates, the terminated child process and the rest of the children started after it, are terminated and restarted.

But there should be no interdependencies between the workers, so `one_for_one` seems more appropriate.

File: lib/ex_schedule.ex [hunk 2]
```diff
@@ -132,8 +132,12 @@ defmodule ExSchedule do
       defp normalize_namespace(ns) when is_bitstring(ns), do: String.to_atom(ns)
       defp normalize_namespace(ns), do: ns
 
-      defp child_spec(schedule) do
-        worker(ExSchedule.ScheduledTask, [schedule], id: schedule.id)
+      defp start(args) do
+        [head | tail] = args
+
+        children = [
+          {ExSchedule.ScheduledTask, head}
```
Comment (zorbash) [inline lib/ex_schedule.ex:139]: This will always return a list with a **single** child spec, but we need to return a child spec spec for each scheduled task.

We could change `child_spec/1` to return the new map child specs with:
 
```elixir
defp child_spec(schedule) do
  %{id: schedule.id, start: {ExSchedule.ScheduledTask, :start_link, [schedule]}}
end
```

**Review by zorbash — APPROVED** [review]

**Review by luizvarela — APPROVED** [review]

**Review by danhawkins — APPROVED** [review]
